Add match

yuriyoleynikov opened the pull request:
Addresses #37

Commit:
-Change MatchController class, add Add Method
-Add PastMatchEvent class
-Add AddMatchViewModel class
-Change GameHandler class, add private void On(PastMatchEvent gameEvent)
-Change enum MatchState, add PastMatchImported in enum
-Add Add view
-Add link "Add match" in _layout

Igorbek commented on src/YoTennis/Controllers/MatchController.cs:
```diff
@@ -28,6 +28,47 @@ public MatchController(UserManager<ApplicationUser> userManager, IMatchListServi
             _matchListService = matchListService;
         }
 
+        public async Task<IActionResult> Add()
```
I'd suggest to rename to something like `AddPastMatch`

yuriyoleynikov replied:
Done

Igorbek commented on src/YoTennis/Controllers/MatchController.cs:
```diff
@@ -28,6 +28,47 @@ public MatchController(UserManager<ApplicationUser> userManager, IMatchListServi
+                    },
+                    new SetModel {
+                        Score = new Score {
+                            FirstPlayer = addMatchViewModel.FirstPlayerSet3, SecondPlayer = addMatchViewModel.SecondPlayerSet3 }
```
Is it always going to be 3 sets? How would you detect that is more?

Igorbek commented on src/YoTennis/Models/AddMatchViewModel.cs:
```diff
@@ -0,0 +1,42 @@
+
+namespace YoTennis.Models
+{
+    public class AddMatchViewModel
```
also like `AddPostMatchViewModel`

yuriyoleynikov replied:
Done

Igorbek commented on src/YoTennis/Models/GameHandler.cs:
```diff
@@ -155,7 +155,23 @@ public void AddEvent(GameEvent gameEvent)
+            CurrentState.SecondPlayerUserId = gameEvent.SecondPlayerUserId;
+
+            CurrentState.Sets = gameEvent.MatchScore;
+
```
the handler does not change state, so it will still be `NotStarted`

yuriyoleynikov replied:
Done

yuriyoleynikov replied:
This was hard!

Igorbek reviewed (changes requested):
In addition to comments, please add tests that covers past match importing

Commit:
-Renaming Add to AddPastMatch
-Renaming AddMatchViewModel to AddPostMatchViewModel
-Change method On in GameHandler
-Change AddEventAsync method in DatabaseMatchService

Commit:
-Change AddPastMatch method in MatchController class
-Change filds in class AddPostMatchViewModel
-Change view AddPastMatch

Commit:
Add 5 sets functionality

Commit:
Change UI for AddMatch view

Commit:
Add test for PastMatchEvent

Commit:
-Add test Cheсk_Not_Expected_PastMatchEvent_StartEvant
-Change GameHandler class, On method for Exeptions
-Change AddPastMatch method in MatchController class
